src/Generators/Shared/SymbolHelpers.cs

```diff
@@ -16,4 +16,40 @@ public static string GetFullNamespace(ISymbol symbol)
     {
         return symbol.ContainingNamespace.IsGlobalNamespace ? string.Empty : symbol.ContainingNamespace.ToString();
     }
+
+    /// <summary>
+    /// Can code in a given type access a given member?
+    /// </summary>
+    /// <remarks>
+    /// Note that this implementation assumes that the target member is within the origin type
+    /// or a base class of the origin type.
+    /// </remarks>
+    public static bool CanAccess(this INamedTypeSymbol originType, ISymbol targetMember)
```

Comment:
@jaredpar Oh great Roslyn oracle!

Is there a canonical way to do this in Roslyn that knows all the accessibility rules and handles IVT? I just want to know whether code located in originType can utter the name of the targetMember.

Reply:
@333fred

Reply:
@geeknoid `Compilation.IsSymbolAccessibleWithin`

Reply:
@geeknoid FYI

Reply:
Applied in #4850